### PR TITLE
REGR: fix inplace arithmetic operation on Series no longer updating parent DataFrame

### LINKED ISSUE #36373: REGR: inplace arithmetic operation on Series no longer updating parent DataFrame
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [X] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd

df = pd.DataFrame({"A": [1, 2, 3]})

a = df["A"]
a += 999

print(df)
print(df["A"])

assert a is df["A"]
```

#### Problem description

I expect `print(df)` and `print(df["A"])` to show identical values for column `A`.  This is not the case.  The output of the code above:
```
   A
0  1
1  2
2  3
0    1000
1    1001
2    1002
Name: A, dtype: int64
```

#### Expected Output
```
   A
0  1000
1  1001
2  1002
0    1000
1    1001
2    1002
Name: A, dtype: int64
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : fd20f7d34b8601a99b450a615241c233f034fc88
python           : 3.8.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-47-generic
Version          : #51-Ubuntu SMP Fri Sep 4 19:50:52 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.0.dev0+379.gfd20f7d34
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None
None

</details>

Possibly related to https://github.com/pandas-dev/pandas/pull/36051?

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #36373

See discussion at https://github.com/pandas-dev/pandas/pull/36498#issuecomment-708578774 for reverting the original fix (https://github.com/pandas-dev/pandas/pull/30501 for #30484)

### CODE REVIEW

**Review by jreback — APPROVED** [review]
lgtm merge on green.

**Comment by jbrockmendel** [comment]
@jorisvandenbossche i have no objection if you're just looking for the quickest available fix, but another option is to update NDFrame._inplace_method:

```
def _inplace_method(self, other, op):
    result = op(self, other)

+    if self.ndim == 1 and result._indexed_same(self) and result.dtype == self.dtype:
+        self._values[:] = result._values
+        return self

    self._reset_cacher()
    [...]
```

i havent run the full test suite on this yet, but it passes the new test this PR adds and one of the two that this PR xfails.

**Comment by jorisvandenbossche** [comment]
> i have no objection if you're just looking for the quickest available fix

That's indeed the case, as I want to have this fixed for 1.1.4, and I am almost off to bed ;-) 
But certainly feel free to push that change here (or in another PR), if it passes the test we can merge that instead.

**Comment by jreback** [comment]
yeah let's just revert this as is to avoid blocking. @jbrockmendel feel free to push your change in a new PR

**Comment by simonjayhawkins** [comment]
> @jbrockmendel feel free to push your change in a new PR

or could add the change to _maybe_cache_changed from #36498. The blockers on that PR weren't related to that change.

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -888,6 +888,7 @@ def test_identity_slice_returns_new_object(self):
         original_series[:3] = [7, 8, 9]
         assert all(sliced_series[:3] == [7, 8, 9])
 
+    @pytest.mark.xfail(reason="accidental fix reverted - GH37497")
     def test_loc_copy_vs_view(self):
```
Comment (jbrockmendel) [inline pandas/tests/indexing/test_loc.py:892]: im inclined to think this test is wrong.

```
orig = DataFrame(zip(range(3), range(3)), columns=["a", "b"])
df = orig.copy()

ser = df.loc[:, "a"]
ser += 2

tm.assert_frame_equal(df, orig)
```

`df.loc[:, "a"]` is a view on `df`, and `+=` is specifically an inplace op (as long as it doesnt result in a dtype change).  I think `df` _should_ be altered.

Comment (jorisvandenbossche) [reply pandas/tests/indexing/test_loc.py:892]: > im inclined to think this test is wrong.

Yep, I had the same feeling: https://github.com/pandas-dev/pandas/pull/36498#issuecomment-719042038

**Comment by jorisvandenbossche** [comment]
Closing in favor of https://github.com/pandas-dev/pandas/pull/37508, which fixes it without reverting the fix in `drop` (thanks @jbrockmendel !)

**Pull request closed without merging**